src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp

```diff
@@ -427,6 +446,159 @@ void runtime::intelgpu::do_max_pool_backprop_operation(cldnn::topology& topology
     topology.add(op_max_pool_backprop);
 }
 
+void runtime::intelgpu::do_avg_pool_backprop_operation(cldnn::topology& topology,
+                                                       const string& delta_name,
+                                                       const Shape& delta_shape,
+                                                       const string& output_name,
+                                                       const Shape& output_shape,
+                                                       const element::Type& output_type,
+                                                       const Shape& win_shape,
+                                                       const Shape& win_stride,
+                                                       const Shape& pad_below,
+                                                       const bool include_padding)
+{
+    const string entry_point_name = "op_avg_pool_backprop_" + output_name;
+    codegen::CodeWriter writer;
+    vector<size_t> gws;
+
+    const Shape delta_data(delta_shape.cbegin() + 2, delta_shape.cend());
+    const Shape output_data(output_shape.cbegin() + 2, output_shape.cend());
+
+    bool is_zero_pad = true;
+    for (auto const& i : pad_below)
+    {
+        if (i != 0)
+        {
+            is_zero_pad = false;
+            break;
+        }
+    }
+
+    size_t win_elems_size = 1;
+    if (include_padding || is_zero_pad)
+    {
+        for (auto const& i : win_shape)
```

Comment:
Perhaps it is better to use `shape_size<Shape>(input0_shape)` as it done in `do_1d_scalar_mul`